Sources/HTMLKit/Internal/Features/Conversion/Converter.swift

```diff
@@ -644,11 +644,11 @@ extension Converter {
             let indent = String(repeating: "\t", count: (node.level - 1) + (preindent ?? 0))
             
             if let text = text {
-                if node.parent?.localName != "pre" {
-                    let cleanText = text.replacingOccurrences(of: "\n", with: "")
-                    "\(indent)\"\(cleanText)\"\n"
-                } else {
+                if node.parent?.localName = "pre" {
                     "\(indent)\"\"\"\n\(text)\"\"\"\n"
+                } else {
+                    let cleanText = text.trimmingCharacters(in: .whitespacesAndNewlines)
+                    "\(indent)\"\(cleanText)\"\n"
```

Comment:
What's the """ in a pre for?

Reply:
Can you give me a sample of code you try to convert, so I can see what's happening here?

Reply:
```html
<body>
  <h1>Header</h1>
  <h1>      Hello      World!     </h1>
  <p>
    Paragraph
  </p>
<pre>struct ContentView: View {
    var body: some View {
        AdaptiveStack {
            Text("Horizontal when there's lots of space")
            Text("but")
            Text("Vertical when space is restricted")
        }
    }
}
</pre>
</body>
```

Reply:
we preserve the whitespace and the new lines in `pre` and remove them in the other tags.
I was reading [this].(https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace)
So, unless it is in `pre` we should remove all the new lines.
But I just noticed the built-in function trims it only from the beginning and end. I need to do one more commit for non-`pre` case.

Reply:
This Html snippet covers the cases well
```html
<!DOCTYPE html>
<html>
<body>
  <h1>Header</h1>
  <h1>      Hello      
  World!     </h1>
  <p>
    Paragraph
  </p>
<pre>private var comment: String? {
            
            guard let comment = node.stringValue else {
                return nil
            }
            
            return comment
        }
</pre>
<p>struct ContentView: View {
    var body: some View {
        AdaptiveStack {
            Text("Horizontal when there's lots of space")
            Text("but")
            Text("Vertical when space is restricted")
        }
    }
}
</p>
</body>
</html>
```

Reply:
In my view this should be the output. What do you think?
```swift
Html {
            Body {
                Heading1 {
                    "Header"
                }
                Heading1 {
                    "Hello        World!"
                }
                Paragraph {
                    "Paragraph"
                }
                PreformattedText {
                    """
        private var comment: String? {
                    
                    guard let comment = node.stringValue else {
                        return nil
                    }
                    
                    return comment
                }
        """
                }
                Paragraph {
                    #"struct ContentView: View {    var body: some View {        AdaptiveStack {            Text("Horizontal when there's lots of space")            Text("but")            Text("Vertical when space is restricted")        }    }}"#
                }
            }
        }
```